apis/fluentbit/v1alpha2/plugins/custom/custom_types.go

```diff
@@ -0,0 +1,49 @@
+package custom
+
+import (
+	"errors"
+	"fmt"
+	"github.com/fluent/fluent-operator/apis/fluentbit/v1alpha2/plugins"
+	"github.com/fluent/fluent-operator/apis/fluentbit/v1alpha2/plugins/params"
+	"strings"
+)
+
+// +kubebuilder:object:generate:=true
+
+// CustomPlugin is used to support filter plugins that are not implemented yet
+type CustomPlugin struct {
+	Config string `json:"config,omitempty"`
+}
+
+func (c *CustomPlugin) Name() string {
+	splits := strings.Split(c.Config, "\n")
+	for _, i := range splits {
+		tmp := strings.Fields(i)
+		if len(tmp) < 2 {
+			continue
+		}
+		if strings.EqualFold(tmp[0], "Name") {
+			return tmp[1]
+		}
+	}
+	return "NONE"
+}
+
+func (a *CustomPlugin) Params(_ plugins.SecretLoader) (*params.KVs, error) {
```

Comment:
Why need to parser the `config`? Why need to skip the `name` field?

Reply:
A custom plugin is also a plugin that implements the Plugin interface. The plugin name is already resolved in the Name method, so there is no need to resolve the Name again.

Reply:
A more reasonable approach.

Add the `name` filed just when the `p.Name()` is not nil
https://github.com/fluent/fluent-operator/blob/master/apis/fluentbit/v1alpha2/clusterinput_types.go#L94

<img width="686" alt="1663571575(1)" src="https://user-images.githubusercontent.com/53003665/190966520-8e35b191-0964-494e-987c-6cd23d959e2f.png">

and make the `customPlugin.Name()` return nil